cocos/2d/framework/ui-renderer.ts

```diff
@@ -402,13 +411,7 @@ export class UIRenderer extends Renderer {
         if (this.node._uiProps.uiComp === this) {
             this.node._uiProps.uiComp = null;
         }
-        this.destroyRenderData();
-        if (this._materialInstances) {
-            for (let i = 0; i < this._materialInstances.length; i++) {
-                const instance = this._materialInstances[i];
-                if (instance) { instance.destroy(); }
-            }
-        }
```

Comment:
Why instances are not destroyed now?

Reply:
The setSharedMaterial function contains an explicit resource destruction operation. Here's the logic:
```
public setSharedMaterial (material: Material | null, index: number, forceUpdate = false): void {
    if (material && material instanceof MaterialInstance) {
        errorID(12012);
    }

    if (!forceUpdate && this._materials[index] === material) return;

    this._materials[index] = material;
    const inst = this._materialInstances[index];
    if (inst) {
        inst.destroy(); // <-- Destroys the material instance
        this._materialInstances[index] = null;
    }
    this._onMaterialModified(index, this._materials[index]);
}
```

Reply:
I see.